### PR TITLE
better error handling, fixes for parallel make, remove generated files from dist

### PR DESCRIPTION
**Pull request by nigoroll**
…ions which don't

### CODE REVIEW

File: autogen.sh [hunk 1]
```diff
@@ -35,10 +35,17 @@ else
 	esac
 fi
 
-set -ex
-
 # check for varnishapi.m4 in custom paths
-dataroot=$(pkg-config --variable=datarootdir varnishapi)
+dataroot=`pkg-config --variable=datarootdir varnishapi 2>/dev/null`
```
Comment (tfheen) [inline autogen.sh]: Really? $() isn't a bashism, it's part of POSIX sh.

File: autogen.sh [hunk 2]
```diff
@@ -35,10 +35,17 @@ else
...
+if [ -z "$dataroot" ] ; then
+	cat >&2 <<'EOF';
```
Comment (tfheen) [inline autogen.sh]: '' and ; are pointless here.

File: autogen.sh (hunk 1 above)
Comment (nigoroll) [inline autogen.sh]: -----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

On 24/05/15 10:58, Tollef Fog Heen wrote:
> In autogen.sh 
> <https://github.com/varnish/libvmod-example/pull/21#discussion_r30951112>:
> 
>> # check for varnishapi.m4 in custom paths -dataroot=$(pkg-config
>> --variable=datarootdir varnishapi) +dataroot=`pkg-config
>> --variable=datarootdir varnishapi 2>/dev/null`
> 
> Really? $() isn't a bashism, it's part of POSIX sh.

thank you, seems like I got stuck way too far in the past.

>> +if [ -z "$dataroot" ] ; then +	cat >&2 <<'EOF';
> 
> '' and ; are pointless here.

; yes.

http://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_07_04

    If any character in word is quoted, the delimiter shall be formed by
performing quote removal on word, and the here-document lines shall not be
expanded.

I'll update the patch, thanks for the review.

Nils
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQEcBAEBCAAGBQJVYaieAAoJEB3Nj1ejhovXFOcH/1sP/xpNYKQLAaJZEEYblrq9
Si0qOWi05m0zGmsn0sLOxiV6/pPOPYRSoKM9SJTvcBaI3bbBiigYSN9iw1SMK24E
eEsFhsYaaJbkSgxhHrsqpNlteMI12CTdgT8FHFbuh2gYZNFG/dOLbc1w93uHYD7P
vT0AqdisLO2/i146X1NgfugABhX2+eLGOUrxni61hhlOQwFnN7TUk/zUTgPveP5n
0ntHWUq8HdAe9xcxn9Jag8JNjpEOZYatN/hItdT+7NISvrLOmxuBUCSGwq0eAxE6
3gM2QSRYHyovoiLZMTkk18m/LrRsU3YA3DzTiRt0MfvO9u1b79XJKkGruWQcN6k=
=gfLz
-----END PGP SIGNATURE-----

**Comment by nigoroll** [comment]
thanks again @tfheen

**Pull request merged**